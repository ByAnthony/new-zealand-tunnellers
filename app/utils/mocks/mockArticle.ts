import { AboutUsArticle } from "../../types/article";

export const mockArticle: AboutUsArticle = {
  id: "about-us",
  title: "Our Story",
  section: [
    { title: "Title 1", text: "This is the first paragraph of our story." },
    { title: "Title 2", text: "This is the second paragraph of our story." },
  ],
  image: [
    {
      file: "our-story.jpg",
      title: null,
      photographer: null,
      reference: null,
      alt: "A picture of our team",
    },
  ],
};
