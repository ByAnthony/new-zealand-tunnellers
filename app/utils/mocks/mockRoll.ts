import { Tunneller } from '@/app/types/roll';
import { mockId } from './mockProfile';

const johnDoe = {
  forename: 'John',
  surname: 'Doe',
};

const alexanderDriscott = {
  forename: 'Alexander',
  surname: 'Driscott',
};

const williamRight = {
  forename: 'William',
  surname: 'Right',
};

const robertLang = {
  forename: 'Robert',
  surname: 'Lang',
};

export const mockDetails: Tunneller = {
  id: mockId,
  name: johnDoe,
  birthDate: '1886',
  deathDate: '1952',
};

export const mockDetailsList: Tunneller[] = [
  mockDetails,
  {
    ...mockDetails,
    id: 1,
    name: alexanderDriscott,
  },
];

export const mockRoll: Record<string, Tunneller[]> = {
  D: [
    mockDetails,
    {
      ...mockDetails,
      id: 1,
      name: alexanderDriscott,
    },
  ],
  L: [
    {
      ...mockDetails,
      id: 2,
      name: robertLang,
    },
  ],
  R: [
    {
      ...mockDetails,
      id: 3,
      name: williamRight,
    },
  ],
};
