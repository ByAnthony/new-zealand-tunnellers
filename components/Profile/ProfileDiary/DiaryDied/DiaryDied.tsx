"use client";

import { Death } from "@/types/tunneller";

import STYLES from "../ProfileDiary.module.scss";

type Props = {
  death: Death | null;
};

export function DiaryDied({ death }: Props) {
  const title = (ageAtDeath: number | null) =>
    ageAtDeath ? <p>{`Died at the age of ${ageAtDeath}`}</p> : <p>Died</p>;

  if (death && death.date) {
    return (
      <>
        <h3>Death</h3>
        <div className={STYLES["fullwidth-main-card"]}>
          {title(death.ageAtDeath)}
          <span>{`${death.date.dayMonth} ${death.date.year}`}</span>
        </div>
      </>
    );
  }
  return null;
}
