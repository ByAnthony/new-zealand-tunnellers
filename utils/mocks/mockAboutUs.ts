import { AboutUsArticle } from "@/types/article";

import {
  mockSectionOne,
  mockSectionTwo,
  mockTitle,
  mockTopImage,
} from "./mockArticle";

export const mockAboutUs: AboutUsArticle = {
  id: "my-awesome-article",
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage],
};
