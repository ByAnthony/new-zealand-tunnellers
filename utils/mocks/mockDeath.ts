import { Death } from "@/types/tunneller";

import { mockDeath } from "./mockTunneller";

export const mockAfterWarDeath: Death = {
  ...mockDeath,
  warInjuriesDeathAfterWar: true,
};
