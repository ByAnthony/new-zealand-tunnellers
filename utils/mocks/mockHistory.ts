import { HistoryChapter } from "@/types/homepage";

export const mockHistory: HistoryChapter[] = [
  {
    id: "where-all-started",
    chapter: 1,
    title: "Where All\\Started",
    image: "where-all-started.jpg",
  },
  {
    id: "the-journey-ahead",
    chapter: 2,
    title: "The Journey\\Ahead",
    image: "the-journey-ahead.jpg",
  },
  {
    id: "discovering-the-war",
    chapter: 3,
    title: "Discovering\\The War",
    image: "discovering-the-war.jpg",
  },
  {
    id: "below-the-surface",
    chapter: 4,
    title: "Below\\The Surface",
    image: "below-the-surface.jpg",
  },
  {
    id: "back-to-the-surface",
    chapter: 5,
    title: "Back\\To The Surface",
    image: "back-to-the-surface.jpg",
  },
  {
    id: "reconversion-of-activities",
    chapter: 6,
    title: "Reconversion\\Of Activities",
    image: "reconversion-of-activities.jpg",
  },
  {
    id: "returnin-home",
    chapter: 7,
    title: "Returning\\Home",
    image: "returning-home.jpg",
  },
];
