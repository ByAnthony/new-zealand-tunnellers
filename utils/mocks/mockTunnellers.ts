import { Tunneller, TunnellerWithFullNameData } from "@/types/tunnellers";

const mockTunnellerData1: TunnellerWithFullNameData = {
  id: 1,
  forename: "John",
  surname: "Doe",
  birthYear: "1886",
  deathYear: "1952",
  fullName: "John Doe",
};

const mockTunnellerData2: TunnellerWithFullNameData = {
  id: 2,
  forename: "Biff",
  surname: "Tanen",
  birthYear: "1897",
  deathYear: null,
  fullName: "William Smith",
};

const mockTunnellerData3: TunnellerWithFullNameData = {
  id: 3,
  forename: "Emmett",
  surname: "Brown",
  birthYear: null,
  deathYear: "1935",
  fullName: "Emmett Brown",
};

const mockTunnellerData4: TunnellerWithFullNameData = {
  id: 4,
  forename: "Marty",
  surname: "McFly",
  birthYear: null,
  deathYear: null,
  fullName: "Marty McFly",
};

export const mockTunnellersData: TunnellerWithFullNameData[] = [
  mockTunnellerData1,
  mockTunnellerData2,
  mockTunnellerData3,
  mockTunnellerData4,
];

const mockTunneller1: Tunneller = {
  id: 1,
  name: { forename: "John", surname: "Doe" },
  birthYear: "1886",
  deathYear: "1952",
  fullName: "John Doe",
};

const mockTunneller2: Tunneller = {
  id: 2,
  name: { forename: "Biff", surname: "Tanen" },
  birthYear: "1897",
  deathYear: null,
  fullName: "William Smith",
};

const mockTunneller3: Tunneller = {
  id: 3,
  name: { forename: "Emmett", surname: "Brown" },
  birthYear: null,
  deathYear: "1935",
  fullName: "Emmett Brown",
};

const mockTunneller4: Tunneller = {
  id: 4,
  name: { forename: "Marty", surname: "McFly" },
  birthYear: null,
  deathYear: null,
  fullName: "Marty McFly",
};

export const mockTunnellers: Record<string, Tunneller[]> = {
  B: [mockTunneller3],
  D: [mockTunneller1],
  M: [mockTunneller4],
  S: [mockTunneller2],
};
