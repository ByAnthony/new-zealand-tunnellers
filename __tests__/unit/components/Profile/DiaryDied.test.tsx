import { render, screen } from "@testing-library/react";

import { DiaryDied } from "@/components/Profile/ProfileDiary/DiaryDied/DiaryDied";
import { mockAfterWarDeath } from "__tests__/unit/utils/mocks/mockDeath";
import { mockDeath } from "__tests__/unit/utils/mocks/mockTunneller";

describe("Death War", () => {
  const component = <DiaryDied death={mockDeath} />;

  test("renders the component correctly", () => {
    const { asFragment } = render(component);

    expect(asFragment()).toMatchSnapshot();
  });

  test("renders age and date when known", () => {
    render(component);

    expect(screen.getByText("Died at the age of 89")).toBeInTheDocument();
    expect(screen.getByText("13 October 1926")).toBeInTheDocument();
  });

  test("renders date when age unknown", () => {
    const mockComponent = (
      <DiaryDied death={{ ...mockDeath, ageAtDeath: null }} />
    );

    render(mockComponent);

    expect(screen.queryByText("Died at the age of 89")).not.toBeInTheDocument();
    expect(screen.getByText("13 October 1926")).toBeInTheDocument();
  });
});

describe("Post War Death", () => {
  const component = <DiaryDied death={mockAfterWarDeath} />;

  test("renders the component correctly", () => {
    const { asFragment } = render(component);

    expect(asFragment()).toMatchSnapshot();
  });

  test("renders age and date when known", () => {
    render(component);

    expect(screen.getByText("Died at the age of 89")).toBeInTheDocument();
    expect(screen.getByText("13 October 1926")).toBeInTheDocument();
  });

  test("renders date when age unknown", () => {
    const mockComponent = (
      <DiaryDied death={{ ...mockAfterWarDeath, ageAtDeath: null }} />
    );

    render(mockComponent);

    expect(screen.queryByText("Died at the age of 89")).not.toBeInTheDocument();
    expect(screen.getByText("13 October 1926")).toBeInTheDocument();
  });
});

test("does not render the component when unknown", () => {
  const { container } = render(<DiaryDied death={null} />);

  expect(container).toBeEmptyDOMElement();
});
