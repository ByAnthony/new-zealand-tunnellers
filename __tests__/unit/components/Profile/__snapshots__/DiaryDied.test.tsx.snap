// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Death War renders the component correctly 1`] = `
<DocumentFragment>
  <h3>
    Death
  </h3>
  <div
    class="fullwidth-main-card"
  >
    <p>
      Died at the age of 89
    </p>
    <span>
      13 October 1926
    </span>
  </div>
</DocumentFragment>
`;

exports[`Post War Death renders the component correctly 1`] = `
<DocumentFragment>
  <h3>
    Death
  </h3>
  <div
    class="fullwidth-main-card"
  >
    <p>
      Died at the age of 89
    </p>
    <span>
      13 October 1926
    </span>
  </div>
</DocumentFragment>
`;
